ath79: some mtd-cal-data removals

neheb opened the pull request:
no idea if this is even right. Completely untested.

DragonBluep commented on target/linux/ath79/dts/qca9531_8dev_lima.dts:
```diff
@@ -80,6 +80,10 @@
 					};
+
+					calibration_art_1000: calibration@1000 {
+						reg = <0x1000 0x800>;
```
All SoC integrated ath9k wireless cal data sizes are `0x440`. Include AR933x, AR934x and QCA95xx series.

DragonBluep replied:
Larger EEPROMs can also be accepted by the ath9k driver. I think the script is indeed incorrect. It doesn't make sense to use different EEPROM sizes for the same chip and the same driver.

neheb replied:
@andreybondar59 can you verify this?

neheb replied:
@pepe2k might be able to as well.

neheb replied:
looking at this again, I think you're right. 

I'll rework this PR and push later.

raenye commented on target/linux/ath79/dts/ar7161_meraki_mr16.dts:
```diff
@@ -84,8 +83,8 @@
 		compatible = "pci168c,0029";
 		reg = <0x9000 0 0 0 0>;
 		qca,no-eeprom;
```
Why delete once but not twice?

neheb replied:
whoops

neheb replied:
Fixed.

raenye commented on target/linux/ath79/dts/ar9132_tplink_tl-wa901nd-v2.dts:
```diff
@@ -91,10 +91,20 @@
 			};
 
-			art: partition@2 {
+			partition@2 {
```
```suggestion
			partition@3f0000 {
```

neheb replied:
Done.

raenye commented on target/linux/ath79/dts/ar9132_tplink_tl-wr1043nd-v1.dts:
```diff
@@ -117,10 +117,20 @@
 			};
 
-			art: partition@7F0000 {
+			partition@7F0000 {
```
```suggestion
			partition@7f0000 {
```
Lowercase?

neheb replied:
Done.

raenye commented on target/linux/ath79/dts/ar9330_dlink_dir-505.dts:
```diff
@@ -101,16 +104,40 @@
+					#size-cells = <1>;
+
+					macaddr_mac_4: macaddr@4 {
+						reg = <0x4 0x11>;
```
Does this work without `mac-base`?

neheb replied:
No idea. ping @rmilecki

neheb replied:
Looks like you're right. Updated.

raenye commented on target/linux/ath79/dts/ar9330_glinet_gl-ar150.dts:
```diff
@@ -101,7 +101,7 @@
 			};
 
-			art: partition@3 {
+			partition@3 {
```
```suggestion
			partition@ff0000 {
```

raenye commented on target/linux/ath79/dts/ar9331_embeddedwireless_dorin.dts:
```diff
@@ -81,7 +81,7 @@
 			};
 
-			art: partition@3 {
+			partition@3 {
```
```suggestion
			partition@ff0000 {
```

neheb replied:
Done.

raenye commented on target/linux/ath79/dts/ar9331_etactica_eg200.dts:
```diff
@@ -106,7 +106,7 @@
 			};
 
-			art: art@ff0000 {
+			art@ff0000 {
```
Should all partitions be called `partition@x`?

neheb replied:
I really don't want to make too many unrelated changes. with this for example, notice how label is missing.

raenye commented on target/linux/ath79/dts/ar9344_qxwlan_e750x.dtsi:
```diff
@@ -99,10 +99,20 @@
-			art: partition@60000 {
+			partition@60000 {
 				label = "art";
 				reg = <0x060000 0x010000>;
```
Maybe drop leading zeros?

neheb replied:
Done.

neheb committed:
ath79: mtd-cal-data removals

Replacement can be done with nvmem.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

neheb committed:
ath79: remove unused dts labels

These are not referenced.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

neheb commented:
Finally rebased.

@DragonBluep can you check if the first commit is correct?

DragonBluep commented on target/linux/ath79/dts/ar9344_pcs_cap324.dts:
```diff
@@ -150,10 +153,10 @@
 &wmac {
 	status = "okay";
 
+	nvmem-cells = <&macaddr_art_0 (-1)>, <&cal_art_5000>;
```
I think the dts file is correct. It seems that wmac always load eeprom from low address `0x1000`. And the PCIe NIC ([AR9382](https://openwrt.org/toh/hwdata/powercloud_systems/powercloud_systems_cap324)) eeprom size should be the same 0x440.

Here is an eeprom collection.
[art.zip](https://github.com/openwrt/openwrt/files/14318197/art.zip)

neheb replied:
The implication is that currently, both the wmac and pcie devices use the same calibration.

DragonBluep replied:
I found an older patch can be referenced.
https://patchwork-proxy.ozlabs.org/project/openwrt/patch/1450392667-22656-2-git-send-email-openwrt@daniel.thecshore.com/

neheb replied:
ping @danielfdickinson

neheb replied:
Removed mtd-cal-data from pcie instead. It's not correct anyway.

danielfdickinson commented:
@neheb It has been a long time, but I remember that the radios on CAP324 were reversed from what one would expect. That is the 5Ghz used what was normally the 2.4Ghz calibration and vice versa.

I'm not sure how much I can help now; I'd have to check if I even still have any of these devices to test and I haven't built or used OpenWrt in a few years.

neheb commented:
Sounds like https://github.com/openwrt/openwrt/commit/a14170b6e9c8bab493c5e0e7689405a8a0675e is wrong then.

neheb commented:
@DragonBluep btrw since you did work on this, know anything about https://github.com/openwrt/openwrt/blob/main/target/linux/ath79/dts/ar9344_dlink_dir-8x5.dtsi#L61 ?

I originally had qca,no-eeprom removals but aborted when I found this. Probably better to keep this PR to just mtd-cal-data removals.

DragonBluep commented:
> @DragonBluep btrw since you did work on this, know anything about https://github.com/openwrt/openwrt/blob/main/target/linux/ath79/dts/ar9344_dlink_dir-8x5.dtsi#L61 ?
> 
> I originally had qca,no-eeprom removals but aborted when I found this. Probably better to keep this PR to just mtd-cal-data removals.

The previous `mac-address-ascii` implementation doesn't work on some platform driver, see https://github.com/openwrt/openwrt/pull/9780#issuecomment-1486188837. It should be fine now since we have ported it to `mac-base` nvmem layout. At least I can sure mac-address nvmem cell works on ar9287 and ar938x ath9k PCIe devices.

neheb committed:
ath79: convert pcs,cap324 calibration to nvmem

Userspace handling is deprecated.

Also fix a bug with userspace handling where the wrong calibration data
was being used for the PCI card. The dts was correct but userspace was
not.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

neheb commented:
OK, made https://github.com/openwrt/openwrt/pull/14666 for that.

neheb commented:
Rebased.

neheb commented:
ping @Ansuel 

This is ready. Problematic changes were split into a separate commit.

DragonBluep commented on target/linux/ath79/dts/ar1022_sitecom_wlr-7100.dts:
```diff
@@ -147,7 +147,7 @@
 			};
 
-			art: partition@7f0000 {
+			partition@7f0000 {
```
Hi! Please move these changes to a different commit.

neheb replied:
hmm? Do you mean the standalone art: removals?

DragonBluep replied:
Yes. This file does not contain nvmem changes, so it should be a separate commit.

neheb replied:
Done.

DragonBluep commented on target/linux/ath79/dts/ar9132_tplink_tl-wa901nd-v2.dts:
```diff
@@ -87,16 +87,26 @@
 			};
 
-			partition@1 {
+			partition@20000 {
```
This should also be moved to a new commit.

neheb replied:
Done.

DragonBluep reviewed (commented):
Would you mind rebase it. I remember recent commits have introduced some new `mtd-cal-data` properties.

neheb committed:
ath79: fix dtc address warnings

These should match reg values.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

DragonBluep commented on target/linux/ath79/dts/qca9563_qxwlan_e1700ac.dtsi:
```diff
@@ -113,6 +113,10 @@
 					#address-cells = <1>;
 					#size-cells = <1>;
 
+					cal_art_1000: calibration@1000 {
```
Move it to nvmem commit.

neheb replied:
Done.

DragonBluep commented on target/linux/ath79/dts/ar7161_buffalo_wzr-hp-ag300h.dtsi:
```diff
@@ -146,7 +146,7 @@
 			};
 
-			art: partition@50000 {
+			partition@50000 {
```
Move it to `label` commit.

neheb replied:
Done.

DragonBluep commented on target/linux/ath79/dts/ar7240_netgear_wnr612-v2.dtsi:
```diff
@@ -101,7 +101,7 @@
 			};
 
-			art: partition@3f0000 {
+			partition@3f0000 {
```
Move it to `label` commit.

neheb replied:
Done.

DragonBluep commented on target/linux/ath79/dts/ar7240_tplink.dtsi:
```diff
@@ -99,7 +99,7 @@
 			};
 
-			art: partition@3f0000 {
+			partition@3f0000 {
```
Move it to `label` commit.

neheb replied:
Done.

DragonBluep commented on target/linux/ath79/dts/ar7242_ubnt_sw.dtsi:
```diff
@@ -100,7 +100,7 @@
 			};
 
-			art: partition@7f0000 {
+			partition@7f0000 {
```
Move it to `label` commit.

neheb replied:
Done.

DragonBluep commented on target/linux/ath79/dts/ar724x_ubnt_xm.dtsi:
```diff
@@ -58,7 +58,7 @@
 			};
 
-			art: partition@7f0000 {
+			partition@7f0000 {
```
Move it to `label` commit.

neheb replied:
Done.

DragonBluep commented on target/linux/ath79/dts/ar9341_engenius_eap300-v2.dts:
```diff
@@ -56,7 +56,8 @@
 
-	mtd-cal-data = <&art 0x1000>;
+	nvmem-cells = <&cal_art_1000>;
+	nvmem-cell-names = "mac-address", "calibration";
```
nit.

DragonBluep commented on target/linux/ath79/dts/ar9342_ubnt_xw.dtsi:
```diff
@@ -62,7 +62,7 @@
 			};
 
-			art: partition@7f0000 {
+			partition@7f0000 {
```
Move it to `label` commit.

DragonBluep commented on target/linux/ath79/dts/ar9342_zyxel_nwa11xx.dtsi:
```diff
@@ -111,7 +111,7 @@
 			};
 
-			art: partition@ff0000 {
+			partition@ff0000 {
```
Move it to `label` commit.

DragonBluep commented on target/linux/ath79/dts/ar9344_compex_wpj344-16m.dts:
```diff
@@ -98,10 +98,28 @@
+
+					macaddr_art_6: macaddr@6 {
+						reg = <0x6 0x6>;
+					};
```
Useless mac address cells?

neheb replied:
Copy/paste error I think.

DragonBluep commented on target/linux/ath79/dts/ar9344_dlink_dir-8x5.dtsi:
```diff
@@ -135,7 +135,7 @@
 			};
 
-			art: partition@ff0000 {
+			partition@ff0000 {
```
Move it to `label` commit.

neheb replied:
Done.

DragonBluep commented on target/linux/ath79/dts/ar9344_qxwlan_e750x.dtsi:
```diff
@@ -101,10 +101,20 @@
+			partition@60000 {
 				label = "art";
-				reg = <0x060000 0x010000>;
+				reg = <0x60000 0x010000>;
```
unnecessary change?

neheb replied:
Moved to first commit.

DragonBluep commented on target/linux/ath79/dts/ar9344_tplink_tl-wdrxxxx.dtsi:
```diff
@@ -103,7 +103,7 @@
 			};
 
-			art: partition@7f0000 {
+			partition@7f0000 {
```
Move it to `label` commit.

neheb replied:
Done.

DragonBluep commented on target/linux/ath79/dts/ar9344_wd_mynet-nxxx.dtsi:
```diff
@@ -58,7 +58,7 @@
 			};
 
-			art: partition@ff0000 {
+			partition@ff0000 {
```
Move it to `label` commit.

neheb replied:
Done.

DragonBluep commented on target/linux/ath79/dts/qca9531_glinet_gl-s200.dtsi:
```diff
@@ -131,7 +131,7 @@
 			};
 
-			art: partition@50000 {
+			partition@50000 {
```
Move it to `label` commit.

neheb replied:
Done.

DragonBluep commented on target/linux/ath79/dts/qca9558_comfast_cf-e380ac-v2.dts:
```diff
@@ -63,12 +63,6 @@
-&eth0 {
-	status = "okay";
-	mtd-mac-address = <&art 0x0>;
-};
```
Move it to a new commit.

neheb replied:
Done.

DragonBluep commented on target/linux/ath79/dts/qca955x_senao_router-dual.dtsi:
```diff
@@ -98,7 +98,7 @@
 			};
 
-			art: partition@ff0000 {
+			partition@ff0000 {
```
Move it to `label` commit.

neheb replied:
Done.

DragonBluep commented on target/linux/ath79/dts/qca955x_ubnt_xc.dtsi:
```diff
@@ -53,7 +53,7 @@
 			};
 
-			art: partition@ff0000 {
+			partition@ff0000 {
```
Move it to `label` commit.

neheb replied:
Done.

DragonBluep commented on target/linux/ath79/dts/qca9563_dlink_covr.dtsi:
```diff
@@ -85,7 +85,7 @@
 			};
 
-			art: partition@ff0000 {
+			partition@ff0000 {
```
Move it to `label` commit.

neheb replied:
Done.

DragonBluep commented on target/linux/ath79/dts/qca9563_dlink_dir-842-c.dtsi:
```diff
@@ -107,7 +107,7 @@
 			};
 
-			art: partition@fd0000 {
+			partition@fd0000 {
```
Move it to `label` commit.

neheb replied:
Done.

DragonBluep reviewed (commented):
Hi! I find some tiny issues.

neheb committed:
ath79: cd-e380ac-v2: remove duplicate eth0

It seems in the transition to nvmem, eth0 was specified twice and was
already converted to use nvmem.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

DragonBluep reviewed (approved):
LGTM.
`Reviewed-by: Shiji Yang <yangshiji66@qq.com>`

neheb commented:
@DragonBluep added last 3 remaining devices. Now git grep returns only package/kernel/mac80211/patches/ath9k/552-ath9k-ahb_of.patch for mtd-cal-data. I can't tell whether or not that can be removed.

DragonBluep commented:
> @DragonBluep added last 3 remaining devices. Now git grep returns only package/kernel/mac80211/patches/ath9k/552-ath9k-ahb_of.patch for mtd-cal-data. I can't tell whether or not that can be removed.

This patch contains a lot of other hacks so we can not remove it.

neheb commented:
Maybe just the mtd-cal-data part?

neheb committed:
ath79: 8dev: remove wmac userspace handling

Before the nvmem rework, this was already handled in dts with
mtd-cal-data instead of qca,no-eeprom. No need to duplicate. Also, the
800 size value seems nonsensical. 440 is the standard.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

neheb commented:
@DragonBluep added extra commit. You mentioned this is nonsensical.

DragonBluep commented:
> Maybe just the mtd-cal-data part?

Let's wait other people's comments. Personally I'd like to remove it because it is not an upstream feature and we don't need it now.

> @DragonBluep added extra commit. You mentioned this is nonsensical.

Still LGTM.

neheb commented:
ping @hauke
